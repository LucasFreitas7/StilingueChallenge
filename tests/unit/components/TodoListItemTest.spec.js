import { shallowMount } from '@vue/test-utils'
import TodoListItem from '../../../src/components/TodoListItem.vue'


const handleClickMock = jest.fn()
const handleClickOnDeleteMock= jest.fn()
describe('TodoListItem.vue', () => {
  const wrapper = shallowMount(TodoListItem, {
    propsData:{
        todo: {
            id:1,
            name: '10 minutes meditation',
            done: false
          }
    }
  })
  it('teste de snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Quando cliclado no Check do Item deve ser disparado a função handleClick', () => {
  
    wrapper.setMethods({handleClick: handleClickMock})
    wrapper.findAll('div').at(1).trigger('click')
    expect(handleClickMock).toHaveBeenCalled()
  })
  it('Quando cliclado no delete do Item deve ser disparado a função handleClickOnDelete', () => {
    wrapper.setMethods({handleClickOnDelete: handleClickOnDeleteMock})
    wrapper.findAll('div').at(2).trigger('click')
    expect(handleClickOnDeleteMock).toHaveBeenCalled()
  })
})
