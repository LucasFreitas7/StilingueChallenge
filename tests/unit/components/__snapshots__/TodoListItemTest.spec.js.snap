// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem.vue teste de snapshot 1`] = `
<div class="list-item">
  <div class="todo-check"></div> <span class="">10 minutes meditation</span>
  <div class="todo-delete"></div>
</div>
`;
